### PR TITLE
31 media search service

### PR DESCRIPTION
**Pull request by this-is-joeking**
Issue #31 

# Summary of things changed:
Added search endpoint GET `/api/v1/media?q=<query>`
Used existing PORO
Created additional serializer so that FE doesn't get a bunch of attributes with value of `nil`
Created methods for facade and service along with applicable tests.

Changes that will also impact the media details response:
- attribute[:type] to attribute[:media_type]

# List any known issues (include relevant code snippets): 
  - Only edge case considered is if search returns no results

# Necessary checkmarks: 
  - [x] All tests are passing 
  - [x] Code will run locally 
  - [x] Confirm self-review 
  
# Json response snippet for search with results:

 GET `/api/v1/media?q=everything%20everywhere`

```
{"data"=>
  [{"id"=>"1516721",
    "type"=>"search_result",
    "attributes"=>
     {"id"=>1516721,
      "title"=>"Everything Everywhere All at Once",
      "media_type"=>"movie",
      "release_year"=>2022,
      "tmdb_id"=>545611,
      "tmdb_type"=>"movie",
      "poster_path"=>"https://cdn.watchmode.com/posters/01516721_poster_w185.jpg"}},
   {"id"=>"1126597",
    "type"=>"search_result",
    "attributes"=>
     {"id"=>1126597,
      "title"=>"Everything, Everything",
      "media_type"=>"movie",
      "release_year"=>2017,
      "tmdb_id"=>417678,
      "tmdb_type"=>"movie",
      "poster_path"=>"https://cdn.watchmode.com/posters/01126597_poster_w185.jpg"}},
   {"id"=>"1435331",
    "type"=>"search_result",
    "attributes"=>
     {"id"=>1435331,
      "title"=>"To Wong Foo, Thanks for Everything! Julie Newmar",
      "media_type"=>"movie",
      "release_year"=>1995,
      "tmdb_id"=>9090,
      "tmdb_type"=>"movie",
      "poster_path"=>"https://cdn.watchmode.com/posters/01435331_poster_w185.jpg"}}
    ]
}
```
- Encountered a few movies without a poster 
```
{"id"=>"2757541",
    "type"=>"search_result",
    "attributes"=>
     {"id"=>2757541,
      "title"=>"Everything Starts With Love",
      "media_type"=>"short_film",
      "release_year"=>2023,
      "tmdb_id"=>1090006,
      "tmdb_type"=>"movie",
      "poster_path"=>nil}}
```

# Json response snippet for search with no results:

 GET `/api/v1/media?q=asdhfasjhdgfliwegf`

```
{
    "data": []
}
```

### CODE REVIEW

File: app/poros/media.rb [hunk 1]
```diff
@@ -3,14 +3,14 @@ class Media
...
-              :type,
+              :media_type,
               :description,
               :genres,
               :release_year,
               :runtime,
               :language,
               :sub_services,
-              :poster,
+              :poster_path,
```
Comment (Pocketzs) [inline app/poros/media.rb:13]: Perfect.  Thank you for the uniform names

File: app/services/media_service.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ def self.details(id)
   end
 
   def self.search(query)
-    response = conn.get('/v1/search/', { search_field: 'name', search_value: query, types: 'tv,movie' })
+    response = conn.get('/v1/autocomplete-search/', { search_field: 'name', search_value: query, search_type: '2' })
```
Comment (Pocketzs) [inline app/services/media_service.rb:8]: I see the switch to the different endpoint

**Review by Pocketzs — APPROVED** [review]
Haven't pulled down and tested, but the changes I've seen are expected
